community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/LuceneBatchInserterIndex.java

```diff
@@ -387,11 +398,21 @@ private File getStoreDir( File dbStoreDir )
         }
         return dir;
     }
-    
+
     @Override
     public void flush()
     {
-        writerModified = true;
+        try
+        {
+            while ( !searcherManager.maybeRefresh() )
+            {
+                LockSupport.parkNanos( 5_000_000 ); // wait for 5ms
```

Comment:
how about `LockSupport.parkNanos( TimeUnit.MILLISECONDS.toNanos( 5 ) );` - directly readable and removes the need for the comment.